core/src/Kokkos_Profiling_ScopedProfileRegion.hpp

```diff
@@ -0,0 +1,48 @@
+//@HEADER
+// ************************************************************************
+//
+//                        Kokkos v. 4.0
+//       Copyright (2022) National Technology & Engineering
+//               Solutions of Sandia, LLC (NTESS).
+//
+// Under the terms of Contract DE-NA0003525 with NTESS,
+// the U.S. Government retains certain rights in this software.
+//
+// Part of Kokkos, under the Apache License v2.0 with LLVM Exceptions.
+// See https://kokkos.org/LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//@HEADER
+
+#ifndef KOKKOSP_SCOPED_PROFILE_REGION_HPP
+#define KOKKOSP_SCOPED_PROFILE_REGION_HPP
+#ifndef KOKKOS_IMPL_PUBLIC_INCLUDE
+#define KOKKOS_IMPL_PUBLIC_INCLUDE
+#define KOKKOS_IMPL_PUBLIC_INCLUDE_PROFILING_SCOPEDPROFILEREGION
+#endif
+
+#include <Kokkos_Macros.hpp>
+#include <impl/Kokkos_Profiling.hpp>
+
+#include <string>
+
+namespace Kokkos::Profiling {
+
+class ScopedProfileRegion {
+ public:
+  ScopedProfileRegion(ScopedProfileRegion const &) = delete;
+  ScopedProfileRegion &operator=(ScopedProfileRegion const &) = delete;
+
+  explicit ScopedProfileRegion(std::string const &name) {
```

Comment:
```suggestion
#if defined(__has_cpp_attribute) && __has_cpp_attribute(nodiscard) >= 201907
  [[nodiscard]]
#endif
  explicit ScopedProfileRegion(std::string const &name) {
```

Reply:
See https://github.com/kokkos/kokkos/blob/a75aa23a1c54ff1afa7c13adf36bcee5694d4aea/core/src/Kokkos_Core.hpp#L221-L227 and https://github.com/kokkos/kokkos/pull/5224 to disallow creating a temporary object.